File: src/main/java/com/nexa/platform/invoicing/infrastructure/persistence/jpa/InvoiceRepository.java
package com.nexa.platform.invoicing.infrastructure.persistence.jpa;

import com.nexa.platform.invoicing.domain.model.Invoice;
import com.nexa.platform.invoicing.domain.model.repositories.InvoiceRepositoryPort;
import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface InvoiceRepository extends JpaRepository<Invoice, Long>, InvoiceRepositoryPort {
    Optional<Invoice> findByTenantIdAndInvoiceNumber(Long tenantId, String invoiceNumber);
    List<Invoice> findByTenantIdOrderByIdAsc(Long tenantId);
    List<Invoice> findByOrderIdOrderByIdAsc(Long orderId);
    List<Invoice> findByTenantIdAndOrderIdOrderByIdAsc(Long tenantId, Long orderId);
}

File: King.Nexa.Platform/Invoicing/Infrastructure/Persistence/EntityFrameworkCore/Repositories/InvoiceRepository.cs
using King.Nexa.Platform.Invoicing.Domain.Model.Aggregates;
using King.Nexa.Platform.Invoicing.Domain.Model.ValueObjects;
using King.Nexa.Platform.Invoicing.Domain.Repositories;
using King.Nexa.Platform.Shared.Infrastructure.Persistence.EntityFrameworkCore.Configuration;
using King.Nexa.Platform.Shared.Infrastructure.Persistence.EntityFrameworkCore.Repositories;
using Microsoft.EntityFrameworkCore;

namespace King.Nexa.Platform.Invoicing.Infrastructure.Persistence.EntityFrameworkCore.Repositories;

public class InvoiceRepository(AppDbContext context) : BaseRepository<Invoice>(context), IInvoiceRepository
{
    public async Task<Invoice?> FindByInvoiceNumberAsync(InvoiceNumber invoiceNumber, CancellationToken cancellationToken = default) =>
        await Context.Invoices.FirstOrDefaultAsync(invoice => invoice.InvoiceNumber == invoiceNumber, cancellationToken);
}
